lib/elixir/lib/enum.ex

```diff
@@ -1737,7 +1737,12 @@ defmodule Enum do
   end
 
   def reverse(enumerable, tail) do
-    reduce(enumerable, to_list(tail), fn(entry, acc) ->
+    do_reverse(enumerable, to_list(tail))
+  end
+
+  @spec do_reverse(t, list) :: list
```

Comment:
It should be clear from the tiny function. Using private specs and comments
as such are not common in elixir source, so we should either follow how it
is done today or revisit the whole code add.

-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*